packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -258,36 +255,47 @@ export class GarbageCollector implements IGarbageCollector {
 		 * Set up the initializer which initializes the GC state from the data in base snapshot. This is done when
 		 * connected in write mode or when GC runs the first time. It sets up all unreferenced nodes from the base
 		 * GC state and updates their inactive or sweep-ready state.
+		 *
+		 * Must only be called if there is a current reference timestamp.
 		 */
 		this.initializeGCStateFromBaseSnapshotP = new LazyPromise<void>(async () => {
```

Comment:
I'm trying to think if this makes more sense to turn into a regular function that takes in `currentReferenceTimestamp`.  I'll play around with it once I finish the review.

Reply:
FYI now that there's only one callsite (after you remove the one in `collectGarbage`), and since this doesn't capture any constructor-time state in the closure, it can be replaced with a simple function `private async initializeGCStateFromBaseSnapshot(currentReferenceTimestamp: number)`.  I made the change and stashed it, maybe I'll put a PR out after you merge.

Reply:
Another reason to make this a lazy promise is to ensure it is re-entrant. Even if its called multiple times, the initialization should only happen once. There can be scenarios where a container keeps on reconnecting and this is called multiple times. Another scenario which is more concerning is when the summarizer client connects and starts GC immediately. There are probably other ways to make it eventual consistent, lazy promise is a simple one.